File: 121 Best Time to Buy and Sell Stock/121 Best Time to Buy and Sell Stock.ts
function maxProfit(prices: number[]): number {
  let minPrice: number = prices[0];
  let maxProfit: number = 0;

  for (const price of prices) {
    if (price < minPrice) {
      minPrice = price;
    }

    maxProfit = Math.max(maxProfit, price - minPrice);
  }

  return maxProfit;
}

File: 121 Best Time to Buy and Sell Stock/121 Best Time to Buy and Sell Stock.swift
class Solution {
  func maxProfit(_ prices: [Int]) -> Int {
    var minPrice: Int = prices[0]
    var maxProfit: Int = 0

    for price in prices {
      if price < minPrice {
        minPrice = price
      }

      maxProfit = max(maxProfit, price - minPrice)
    }  

    return maxProfit
  }
}